Source/WebGPU/WGSL/Parser.h

```diff
@@ -27,12 +27,13 @@
 
 #include "ASTShaderModule.h"
 #include "CompilationMessage.h"
-#include "Lexer.h"
 #include <wtf/Expected.h>
 
 namespace WGSL {
 
-Expected<AST::ShaderModule, Error> parseLChar(const String& wgsl);
-Expected<AST::ShaderModule, Error> parseUChar(const String& wgsl);
+struct Configuration;
+
+Expected<AST::ShaderModule, Error> parseLChar(const String& wgsl, const Configuration&);
+Expected<AST::ShaderModule, Error> parseUChar(const String& wgsl, const Configuration&);
```

Comment:
The _parser_ needs the configuration? Really?

I would expect _parsing_ to not be sensitive to the device we're running on...

Reply:
It doesn't, I added a FIXME about that. We need a new data structure that will hold both the AST and the configuration (and potentially more data in the future). I'll fix it soon, but for now I just put it in the module.